### PR TITLE
mutt: 2.2.5 -> 2.2.6

### PR DESCRIPTION
**Pull request by trofi**
While at it added trivial updater script.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by trofi** [comment]
Some `sage / sagedoc` recursion.

**Comment by trofi** [comment]
@ofborg eval

File: pkgs/applications/networking/mailreaders/mutt/default.nix [hunk 1]
```diff
@@ -14,6 +14,7 @@
 , imapSupport  ? true
 , withSidebar  ? true
 , gssSupport   ? true
+, writeScript
 }:
 
 assert headerCache  -> gdbm       != null;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/mutt/default.nix:20]: We should drop the asserts.

File: pkgs/applications/networking/mailreaders/mutt/default.nix [hunk 2]
```diff
@@ -87,6 +88,20 @@ stdenv.mkDerivation rec {
     echo "source $out/etc/gpg.rc" >> $out/etc/Muttrc
   '';
 
+  passthru = {
+    updateScript = writeScript "update-mutt" ''
+      #!/usr/bin/env nix-shell
+      #!nix-shell -i bash -p curl pcre common-updater-scripts
+
+      set -eu -o pipefail
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/mutt/default.nix:96]:
```suggestion
      set -euo pipefail
```

File: pkgs/applications/networking/mailreaders/mutt/default.nix (hunk 1 above)
Comment (trofi) [reply pkgs/applications/networking/mailreaders/mutt/default.nix]: Dropped the asserts and removed top level `with lib;`.

**Review by SuperSandro2000 — APPROVED** [review]

**Review by rnhmjoj — APPROVED** [review]
I haven't tested the update script, but the rest looks ok.

**Pull request merged**